### DISCUSSION TITLE
Trying to build eepp from source without MSYS2

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/eepp#48] [Q&A] [answered]
Without MSYS2, there is only a standalone MinGW-w64 distro and Windows Command Prompt. There will be no POSIX utilities and installable libraries like when using MSYS2, so it will need to build all of the needed dependencies from the included sources in src/thirdparty. This is presumed to be a supported setup. But in practice, it doesn't work.

I get a standalone premake5 from their website. `premake5 gmake2` ran without problems. `mingw32-make SHELL=cmd CC=gcc config=release_x86_64 -C make/windows` ran for a while and failed with this error:

```
"==== Building imageresampler-static (release_x86_64) ===="
Creating ../../obj/windows/x86_64/release/imageresampler-static
"resampler.cpp"
../../src/thirdparty/imageresampler/resampler.cpp:333:28: warning: unused variable 'KAISER_ALPHA' [-Wunused-const-variable]
  333 | static const Resample_Real KAISER_ALPHA = 4.0;
      |                            ^~~~~~~~~~~~
1 warning generated.
Linking imageresampler-static
"==== Building mojoal-static (release_x86_64) ===="
Creating ../../obj/windows/x86_64/release/mojoal-static
"mojoal.c"
../../src/thirdparty/mojoAL/mojoal.c:64:10: fatal error: 'SDL2/SDL.h' file not found
   64 | #include <SDL2/SDL.h>
      |          ^~~~~~~~~~~~
1 error generated.
mingw32-make[1]: *** [mojoal-static.make:169: ../../obj/windows/x86_64/release/mojoal-static/mojoal.o] Error 1
mingw32-make: *** [Makefile:402: mojoal-static] Error 2
mingw32-make: Leaving directory 'C:/Work/eepp/make/windows'
```

The build order of the dependencies are wrong. SDL2 needs to be built before mojoAL.

I tried to read `premake5.lua` but I can't understand the way how it will get SDL2. It will download a prebuilt binary from SDL2's website or will it build from source? The source code for SDL2 is also included in src/thirdparty.

**Comment by SpartanJ** [comment, marked as answer]
You just did not use the right command line parameters for auto-installing SDL2. Please follow the instructions in the `README.md` from this repository [here](https://github.com/SpartanJ/eepp?tab=readme-ov-file#mingw).

**Reply by ghost** [reply]
Please tell me if the ecode release for Windows is built using MSVC or MinGW. Thank you.

**Reply by SpartanJ** [reply]
It's cross-built from Linux using mingw, I don't build it on Windows. 😅
